src/files/FilesPage.js

```diff
@@ -58,28 +55,13 @@ const FilesPage = ({
   */
 
   const onDownload = async (files) => {
-    if (downloadProgress !== null) {
-      return downloadAbort()
-    }
-
-    const { url, filename, method } = await doFilesDownloadLink(files)
-
-    if (method === 'GET') {
-      const link = document.createElement('a')
-      link.href = url
-      link.click()
-    } else {
-      const updater = (v) => setDownloadProgress(v)
-      const { abort } = await downloadFile(url, filename, updater, method)
-      setDownloadAbort(() => abort)
-    }
+    const url = await doFilesDownloadLink(files)
+    const link = document.createElement('a')
+    link.href = url
+    link.click()
```

Comment:
non-blocking Q: instead of creating an element and then programmatically clicking on it can we have the thing the users click have the appropriate URL?

Reply:
> Note: I do agree that this is a significant improvement, and resolves https://github.com/ipfs/ipfs-webui/issues/1887 but I'm wondering if there is a way we can keep the ability to cancel a download, or display download progress. Fine with a separate issue for those items

Downloads will be handled by the browser UI. The user can just cancel it through the browser UI. We only had the option to cancel and display progress on the Web UI itself because we were storing **every single** byte of data in-memory before downloading.

Reply:
> non-blocking Q: instead of creating an element and then programmatically clicking on it can we have the thing the users click have the appropriate URL?

Yes and no. If you select multiple files, what link do you put? You can't as we calculate the CID dynamically.

Reply:
Can't we just redirect the users to download? it will prompt the user to download and not go away from current page too?

Reply:
It doesn't go away from the current page AFAIK. At least not for me?

Reply:
This is the behaviour of the current PR:

https://user-images.githubusercontent.com/5447088/201318295-aa0d3d56-d6d7-43a7-ba6b-d8e15022bc30.mp4